OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -95,8 +99,11 @@ public override bool Tick(Actor self)
 			if (!targetIsHiddenActor && target.Type == TargetType.Actor)
 			{
 				lastVisibleTarget = Target.FromTargetPositions(target);
-				lastVisibleMaximumRange = attackTraits.Where(x => !x.IsTraitDisabled)
-					.Min(x => x.GetMaximumRangeVersusTarget(target));
+				var activeAttackTraits = attackTraits.Where(x => !x.IsTraitDisabled);
```

Comment:
We shouldn't be reaching this branch at all, and adding the guard here is going to cause trouble for #17227.
Bail out in the check at the start.